internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -77,6 +78,50 @@ func createMountsConfig(ctx context.Context, coi *createOptionsInternal) (*mount
 	return &config, nil
 }
 
+// HandleCPULimits handles the logic of converting and validating the CPU limits specified
+// in the OCI spec.
+func HandleCPULimits(ctx context.Context, cid, vmid string, spec *specs.Spec, hostCPUCount, uvmCPUCount int32) (int32, int32, int32, error) {
+	cpuNumSet := 0
+	cpuCount := oci.ParseAnnotationsCPUCount(ctx, spec, oci.AnnotationContainerProcessorCount, 0)
+	if cpuCount > 0 {
+		cpuNumSet++
+	}
+
+	cpuLimit := oci.ParseAnnotationsCPULimit(ctx, spec, oci.AnnotationContainerProcessorLimit, 0)
+	if cpuLimit > 0 {
+		cpuNumSet++
+	}
+
+	cpuWeight := oci.ParseAnnotationsCPUWeight(ctx, spec, oci.AnnotationContainerProcessorWeight, 0)
+	if cpuWeight > 0 {
+		cpuNumSet++
+	}
+
+	if cpuNumSet > 1 {
+		return 0, 0, 0, fmt.Errorf("invalid spec - Windows Process Container CPU Count: '%d', Limit: '%d', and Weight: '%d' are mutually exclusive", cpuCount, cpuLimit, cpuWeight)
+	} else if cpuNumSet == 1 {
+		// usableCPUCount is the number of processors present in whatever environment
+		// the container is running in. It will be either the processor count of the
+		// host, or the UVM, based on if the container is process or hypervisor isolated.
+		usableCPUCount := hostCPUCount
```

Comment:
We only use one of `hostCPUCount` and `uvmCPUCount`. For the purposes of this function, seems clearer to just pass in a single value called `maxCPUCount`. The caller can choose which value to pass in.